File: Card_Select.py
import random
Max_Karten = 51
Zeiten_dict = {i: 0 for i in range(1, Max_Karten + 1)}

def Wahrscheinlichkeitsalgorythmus():
    List_Karten = list(range(1, Max_Karten + 1))
    Wahrscheinlichkeit = [0.0196078431] * Max_Karten
    summe = sum(Wahrscheinlichkeit)
    Karte = random.choices(List_Karten, Wahrscheinlichkeit, k=1) [0]
    return Karte

def Karteziehen():
    Random_Karte = Wahrscheinlichkeitsalgorythmus()
    return Random_Karte

def hochzealen(i):
    global Zeiten_dict
    Nummer = i + 1
    Zeiten_dict[Nummer] = 1

    for key in Zeiten_dict:
        if Zeiten_dict[key] >= 1:
            Zeiten_dict[key] += 1
        if Zeiten_dict[key] > 51:
            Zeiten_dict[key] = 0


for _ in range (1, 1000):
    Random_Karte = Karteziehen()
    hochzealen(Random_Karte)
print(Zeiten_dict)

File: Karten_Spiel_Unity/Assets/Card_Select.cs
using System;
using System.Collections;
using System.Collections.Generic;
using System.Linq;
using UnityEngine;

public class Card_Select : MonoBehaviour
{
    public const int Max_Karten = 51;
    public List<int> Zeiten_dict = new List<int>(new int[Max_Karten]);
    void Start()
    {
        Debug.Log(string.Join(", ", Zeiten_dict));
        int Random_Karte = Wahrscheinlichkeitsalgorythmus(Zeiten_dict);
        hochzeahlen(Random_Karte, Zeiten_dict);
    }

    void Update()
    {

    }

    Dictionary<int, float> berechne_Wahrscheinlichkeit(List<int> Zeiten_dict)
    {
        float k = 0.1f;

        Dictionary<int, float> Gewichtungen = Zeiten_dict
            .Select((value, key) => new { key, value = Mathf.Exp(k * value) })
            .ToDictionary(x => x.key, x => x.value);

        float summe_gewichtungen = Gewichtungen.Values.Sum();

        Dictionary<int, float> Wahrscheinlichkeiten = Gewichtungen
            .ToDictionary(x => x.Key, x => x.Value / summe_gewichtungen);

        return Wahrscheinlichkeiten;
    }

    int Wahrscheinlichkeitsalgorythmus(List<int> Zeiten_dict)
    {
        List<int> List_Karten = Zeiten_dict.Select((value, index) => index).ToList();
        Dictionary<int, float> Wahrscheinlichkeiten = berechne_Wahrscheinlichkeit(Zeiten_dict);
        int Random_Karte = List_Karten.OrderBy(x => UnityEngine.Random.value).First();
        Debug.Log(Random_Karte);
        return Random_Karte;
    }

    void hochzealen(Random_Karte, Zeiten_dict)
    {
        Zeiten_dict[Random_Karte] = 1;

        for key in Zeiten_dict:
        {
            if key != Random_Karte:
            {
                Zeiten_dict[key] += 1;
            }
            else:
            {
                Zeiten_dict[key] = 0;
            }
        }
    }
}